### DISCUSSION TITLE
使用“sealos run"安装k8s集群失败？

### DISCUSSION THREAD

**Discussion by abelit** [labring/sealos#4244] [Q&A] [answered]
我也遇到同样的问题，试了多套环境都卡在这里。求助一下，sealos run在部署k8s集群时拉取“kube-proxy","kube-apiserver","etcd", "kube-controller-manager"等镜像不是拉去sealos构建在本地的，我经过多次测试，发现还是拉取“registry.k8s.io”下面的镜像，所以镜像拉取不成功。请问有什么办法真正的做到使用sealos离线部署k8s集群，特别是解决拉取“kube-proxy","kube-apiserver","etcd", "kube-controller-manager"等镜像能够做到离线。
下面是我遇到的报错信息：
2023-11-03T14:25:35 info Executing pipeline Init in CreateProcessor.
2023-11-03T14:25:35 info start to copy kubeadm config to master0
2023-11-03T14:25:35 info start to generate cert and kubeConfig..., 1446 it/s)
2023-11-03T14:25:35 info start to generator cert and copy to masters...
2023-11-03T14:25:35 info apiserver altNames : {map[apiserver.cluster.local:apiserver.cluster.local kubernetes:kubernetes kubernetes.default:kubernetes.default kubernetes.default.svc:kubernetes.default.svc kubernetes.default.svc.cluster.local:kubernetes.default.svc.cluster.local localhost:localhost node1:node1] map[10.10.12.210:10.10.12.210 10.10.12.211:10.10.12.211 10.10.12.212:10.10.12.212 10.103.97.2:10.103.97.2 10.96.0.1:10.96.0.1 127.0.0.1:127.0.0.1]}
2023-11-03T14:25:35 info Etcd altnames : {map[localhost:localhost node1:node1] map[10.10.12.210:10.10.12.210 127.0.0.1:127.0.0.1 ::1:::1]}, commonName : node1
2023-11-03T14:25:35 info sa.key sa.pub already exist
2023-11-03T14:25:36 info start to copy etc pki files to masters
2023-11-03T14:25:36 info start to copy etc pki files to masters                        
2023-11-03T14:25:36 info start to create kubeconfig...                                 
2023-11-03T14:25:37 info start to copy kubeconfig files to masters
2023-11-03T14:25:37 info start to copy static files to masters1/1, 1712 it/s)
2023-11-03T14:25:37 info start to init master0...
10.10.12.210:22	2023-11-03T14:25:37 info domain apiserver.cluster.local:10.10.12.210 append success
10.10.12.210:22	W1103 14:25:37.537686   10676 initconfiguration.go:307] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeproxy.config.k8s.io", Version:"v1alpha1", Kind:"KubeProxyConfiguration"}: strict decoding error: unknown field "udpIdleTimeout"
10.10.12.210:22	W1103 14:25:37.538138   10676 configset.go:177] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeproxy.config.k8s.io", Version:"v1alpha1", Kind:"KubeProxyConfiguration"}: strict decoding error: unknown field "udpIdleTimeout"
10.10.12.210:22	W1103 14:25:37.539079   10676 initconfiguration.go:120] Usage of CRI endpoints without URL scheme is deprecated and can cause kubelet errors in the future. Automatically prepending scheme "unix" to the "criSocket" with value "/run/containerd/containerd.sock". Please update your configuration!
10.10.12.210:22	W1103 14:25:37.539099   10676 utils.go:69] The recommended value for "healthzBindAddress" in "KubeletConfiguration" is: 127.0.0.1; the provided value is: 0.0.0.0
10.10.12.210:22	[init] Using Kubernetes version: v1.28.2
10.10.12.210:22	[preflight] Running pre-flight checks
10.10.12.210:22	[preflight] Pulling images required for setting up a Kubernetes cluster
10.10.12.210:22	[preflight] This might take a minute or two, depending on the speed of your internet connection
10.10.12.210:22	[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
10.10.12.210:22	W1103 14:35:52.321039   10676 checks.go:835] detected that the sandbox image "sealos.hub:5000/pause:3.9" of the container runtime is inconsistent with that used by kubeadm. It is recommended that using "registry.k8s.io/pause:3.9" as the CRI sandbox image.
10.10.12.210:22	error execution phase preflight: [preflight] Some fatal errors occurred:
10.10.12.210:22		[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-apiserver:v1.28.2: output: E1103 14:28:11.386204   10793 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-apiserver/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout" image="registry.k8s.io/kube-apiserver:v1.28.2"
10.10.12.210:22	time="2023-11-03T14:28:11+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-apiserver/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout"
10.10.12.210:22	, error: exit status 1
10.10.12.210:22		[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-controller-manager:v1.28.2: output: E1103 14:30:44.747151   10865 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-controller-manager/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout" image="registry.k8s.io/kube-controller-manager:v1.28.2"
10.10.12.210:22	time="2023-11-03T14:30:44+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-controller-manager/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout"
10.10.12.210:22	, error: exit status 1
10.10.12.210:22		[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-scheduler:v1.28.2: output: E1103 14:33:19.180769   10932 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-scheduler/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout" image="registry.k8s.io/kube-scheduler:v1.28.2"
10.10.12.210:22	time="2023-11-03T14:33:19+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-scheduler/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout"
10.10.12.210:22	, error: exit status 1
10.10.12.210:22		[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-proxy:v1.28.2: output: E1103 14:35:52.304795   11001 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-proxy/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout" image="registry.k8s.io/kube-proxy:v1.28.2"
10.10.12.210:22	time="2023-11-03T14:35:52+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-proxy/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout"
10.10.12.210:22	, error: exit status 1
10.10.12.210:22		[ERROR ImagePull]: failed to pull image registry.k8s.io/pause:3.9: output: E1103 14:38:25.371135   11081 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/pause:3.9\": failed to resolve reference \"registry.k8s.io/pause:3.9\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/pause/manifests/3.9\": dial tcp 64.233.188.82:443: i/o timeout" image="registry.k8s.io/pause:3.9"
10.10.12.210:22	time="2023-11-03T14:38:25+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/pause:3.9\": failed to resolve reference \"registry.k8s.io/pause:3.9\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/pause/manifests/3.9\": dial tcp 64.233.188.82:443: i/o timeout"
10.10.12.210:22	, error: exit status 1
10.10.12.210:22		[ERROR ImagePull]: failed to pull image registry.k8s.io/etcd:3.5.9-0: output: E1103 14:41:00.878407   11149 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/etcd:3.5.9-0\": failed to resolve reference \"registry.k8s.io/etcd:3.5.9-0\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/etcd/manifests/3.5.9-0\": dial tcp 64.233.188.82:443: i/o timeout" image="registry.k8s.io/etcd:3.5.9-0"
10.10.12.210:22	time="2023-11-03T14:41:00+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/etcd:3.5.9-0\": failed to resolve reference \"registry.k8s.io/etcd:3.5.9-0\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/etcd/manifests/3.5.9-0\": dial tcp 64.233.188.82:443: i/o timeout"
10.10.12.210:22	, error: exit status 1
10.10.12.210:22		[ERROR ImagePull]: failed to pull image registry.k8s.io/coredns/coredns:v1.10.1: output: E1103 14:43:35.063878   11217 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to resolve reference \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/coredns/coredns/manifests/v1.10.1\": dial tcp 74.125.203.82:443: i/o timeout" image="registry.k8s.io/coredns/coredns:v1.10.1"
10.10.12.210:22	time="2023-11-03T14:43:35+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to resolve reference \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/coredns/coredns/manifests/v1.10.1\": dial tcp 74.125.203.82:443: i/o timeout"
10.10.12.210:22	, error: exit status 1
10.10.12.210:22	[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
10.10.12.210:22	To see the stack trace of this error execute with --v=5 or higher
2023-11-03T14:43:35 error Applied to cluster error: failed to init init master0 failed, error: run command `kubeadm init --config=/root/.sealos/default/etc/kubeadm-init.yaml --skip-certificate-key-print --skip-token-print -v 0 --ignore-preflight-errors=SystemVerification` on 10.10.12.210:22, output: W1103 14:25:37.537686   10676 initconfiguration.go:307] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeproxy.config.k8s.io", Version:"v1alpha1", Kind:"KubeProxyConfiguration"}: strict decoding error: unknown field "udpIdleTimeout"
W1103 14:25:37.538138   10676 configset.go:177] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeproxy.config.k8s.io", Version:"v1alpha1", Kind:"KubeProxyConfiguration"}: strict decoding error: unknown field "udpIdleTimeout"
W1103 14:25:37.539079   10676 initconfiguration.go:120] Usage of CRI endpoints without URL scheme is deprecated and can cause kubelet errors in the future. Automatically prepending scheme "unix" to the "criSocket" with value "/run/containerd/containerd.sock". Please update your configuration!
W1103 14:25:37.539099   10676 utils.go:69] The recommended value for "healthzBindAddress" in "KubeletConfiguration" is: 127.0.0.1; the provided value is: 0.0.0.0
[init] Using Kubernetes version: v1.28.2
[preflight] Running pre-flight checks
[preflight] Pulling images required for setting up a Kubernetes cluster
[preflight] This might take a minute or two, depending on the speed of your internet connection
[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
W1103 14:35:52.321039   10676 checks.go:835] detected that the sandbox image "sealos.hub:5000/pause:3.9" of the container runtime is inconsistent with that used by kubeadm. It is recommended that using "registry.k8s.io/pause:3.9" as the CRI sandbox image.
error execution phase preflight: [preflight] Some fatal errors occurred:
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-apiserver:v1.28.2: output: E1103 14:28:11.386204   10793 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-apiserver/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout" image="registry.k8s.io/kube-apiserver:v1.28.2"
time="2023-11-03T14:28:11+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-apiserver/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-controller-manager:v1.28.2: output: E1103 14:30:44.747151   10865 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-controller-manager/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout" image="registry.k8s.io/kube-controller-manager:v1.28.2"
time="2023-11-03T14:30:44+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-controller-manager/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-scheduler:v1.28.2: output: E1103 14:33:19.180769   10932 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-scheduler/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout" image="registry.k8s.io/kube-scheduler:v1.28.2"
time="2023-11-03T14:33:19+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-scheduler/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-proxy:v1.28.2: output: E1103 14:35:52.304795   11001 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-proxy/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout" image="registry.k8s.io/kube-proxy:v1.28.2"
time="2023-11-03T14:35:52+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-proxy/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/pause:3.9: output: E1103 14:38:25.371135   11081 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/pause:3.9\": failed to resolve reference \"registry.k8s.io/pause:3.9\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/pause/manifests/3.9\": dial tcp 64.233.188.82:443: i/o timeout" image="registry.k8s.io/pause:3.9"
time="2023-11-03T14:38:25+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/pause:3.9\": failed to resolve reference \"registry.k8s.io/pause:3.9\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/pause/manifests/3.9\": dial tcp 64.233.188.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/etcd:3.5.9-0: output: E1103 14:41:00.878407   11149 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/etcd:3.5.9-0\": failed to resolve reference \"registry.k8s.io/etcd:3.5.9-0\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/etcd/manifests/3.5.9-0\": dial tcp 64.233.188.82:443: i/o timeout" image="registry.k8s.io/etcd:3.5.9-0"
time="2023-11-03T14:41:00+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/etcd:3.5.9-0\": failed to resolve reference \"registry.k8s.io/etcd:3.5.9-0\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/etcd/manifests/3.5.9-0\": dial tcp 64.233.188.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/coredns/coredns:v1.10.1: output: E1103 14:43:35.063878   11217 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to resolve reference \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/coredns/coredns/manifests/v1.10.1\": dial tcp 74.125.203.82:443: i/o timeout" image="registry.k8s.io/coredns/coredns:v1.10.1"
time="2023-11-03T14:43:35+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to resolve reference \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/coredns/coredns/manifests/v1.10.1\": dial tcp 74.125.203.82:443: i/o timeout"
, error: exit status 1
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher
, error: Process exited with status 1,. Please clean and reinstall
Error: failed to init init master0 failed, error: run command `kubeadm init --config=/root/.sealos/default/etc/kubeadm-init.yaml --skip-certificate-key-print --skip-token-print -v 0 --ignore-preflight-errors=SystemVerification` on 10.10.12.210:22, output: W1103 14:25:37.537686   10676 initconfiguration.go:307] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeproxy.config.k8s.io", Version:"v1alpha1", Kind:"KubeProxyConfiguration"}: strict decoding error: unknown field "udpIdleTimeout"
W1103 14:25:37.538138   10676 configset.go:177] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeproxy.config.k8s.io", Version:"v1alpha1", Kind:"KubeProxyConfiguration"}: strict decoding error: unknown field "udpIdleTimeout"
W1103 14:25:37.539079   10676 initconfiguration.go:120] Usage of CRI endpoints without URL scheme is deprecated and can cause kubelet errors in the future. Automatically prepending scheme "unix" to the "criSocket" with value "/run/containerd/containerd.sock". Please update your configuration!
W1103 14:25:37.539099   10676 utils.go:69] The recommended value for "healthzBindAddress" in "KubeletConfiguration" is: 127.0.0.1; the provided value is: 0.0.0.0
[init] Using Kubernetes version: v1.28.2
[preflight] Running pre-flight checks
[preflight] Pulling images required for setting up a Kubernetes cluster
[preflight] This might take a minute or two, depending on the speed of your internet connection
[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
W1103 14:35:52.321039   10676 checks.go:835] detected that the sandbox image "sealos.hub:5000/pause:3.9" of the container runtime is inconsistent with that used by kubeadm. It is recommended that using "registry.k8s.io/pause:3.9" as the CRI sandbox image.
error execution phase preflight: [preflight] Some fatal errors occurred:
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-apiserver:v1.28.2: output: E1103 14:28:11.386204   10793 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-apiserver/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout" image="registry.k8s.io/kube-apiserver:v1.28.2"
time="2023-11-03T14:28:11+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-apiserver:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-apiserver/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-controller-manager:v1.28.2: output: E1103 14:30:44.747151   10865 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-controller-manager/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout" image="registry.k8s.io/kube-controller-manager:v1.28.2"
time="2023-11-03T14:30:44+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-controller-manager:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-controller-manager/manifests/v1.28.2\": dial tcp 142.251.8.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-scheduler:v1.28.2: output: E1103 14:33:19.180769   10932 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-scheduler/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout" image="registry.k8s.io/kube-scheduler:v1.28.2"
time="2023-11-03T14:33:19+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-scheduler:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-scheduler/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/kube-proxy:v1.28.2: output: E1103 14:35:52.304795   11001 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-proxy/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout" image="registry.k8s.io/kube-proxy:v1.28.2"
time="2023-11-03T14:35:52+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/kube-proxy:v1.28.2\": failed to resolve reference \"registry.k8s.io/kube-proxy:v1.28.2\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/kube-proxy/manifests/v1.28.2\": dial tcp 173.194.174.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/pause:3.9: output: E1103 14:38:25.371135   11081 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/pause:3.9\": failed to resolve reference \"registry.k8s.io/pause:3.9\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/pause/manifests/3.9\": dial tcp 64.233.188.82:443: i/o timeout" image="registry.k8s.io/pause:3.9"
time="2023-11-03T14:38:25+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/pause:3.9\": failed to resolve reference \"registry.k8s.io/pause:3.9\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/pause/manifests/3.9\": dial tcp 64.233.188.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/etcd:3.5.9-0: output: E1103 14:41:00.878407   11149 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = Unknown desc = failed to pull and unpack image \"registry.k8s.io/etcd:3.5.9-0\": failed to resolve reference \"registry.k8s.io/etcd:3.5.9-0\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/etcd/manifests/3.5.9-0\": dial tcp 64.233.188.82:443: i/o timeout" image="registry.k8s.io/etcd:3.5.9-0"
time="2023-11-03T14:41:00+08:00" level=fatal msg="pulling image: failed to pull and unpack image \"registry.k8s.io/etcd:3.5.9-0\": failed to resolve reference \"registry.k8s.io/etcd:3.5.9-0\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/etcd/manifests/3.5.9-0\": dial tcp 64.233.188.82:443: i/o timeout"
, error: exit status 1
	[ERROR ImagePull]: failed to pull image registry.k8s.io/coredns/coredns:v1.10.1: output: E1103 14:43:35.063878   11217 remote_image.go:180] "PullImage from image service failed" err="rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to resolve reference \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/coredns/coredns/manifests/v1.10.1\": dial tcp 74.125.203.82:443: i/o timeout" image="registry.k8s.io/coredns/coredns:v1.10.1"
time="2023-11-03T14:43:35+08:00" level=fatal msg="pulling image: rpc error: code = DeadlineExceeded desc = failed to pull and unpack image \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to resolve reference \"registry.k8s.io/coredns/coredns:v1.10.1\": failed to do request: Head \"https://us-west2-docker.pkg.dev/v2/k8s-artifacts-prod/images/coredns/coredns/manifests/v1.10.1\": dial tcp 74.125.203.82:443: i/o timeout"
, error: exit status 1
[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher
, error: Process exited with status 1,. Please clean and reinstall

**Comment by cdmonkey** [comment]
我也是这个问题，按照离线安装的步骤一步一步操作，最后还是需要下载镜像

**Comment by cdmonkey** [comment, marked as answer]
<img width="788" alt="image" src="https://github.com/labring/sealos/assets/9833686/0e803210-6d18-4eab-a63e-848554f80a49">

请问这个操作有什么问题吗？

**Reply by bxy4543** [reply]
离线安装的关键在于 registry和image-cri-shim，
```
systemctl status registry
systemctl status image-cri-shim
du -sh /var/lib/registry
```
先排除下这些点

**Reply by cdmonkey** [reply]
好的谢谢，我看一下

**Reply by abelit** [reply]
尝试了一下，sealos最新版本（4.3.7）只能支持kubernetes v1.27.7及以下的版本（我测试了kuberntes v1.26.3 v1.27.3 v1.27.7都可以离线安装），目前还不支持kubernetes v1.28.x版本的离线安装。
